certifiers/provider.go

```diff
@@ -0,0 +1,125 @@
+package certifiers
+
+import (
+	certerr "github.com/tendermint/tendermint/certifiers/errors"
+)
+
+// Provider is used to get more validators by other means
+//
+// Examples: MemProvider, files.Provider, client.Provider....
+type Provider interface {
+	// StoreCommit saves a FullCommit after we have verified it,
+	// so we can query for it later. Important for updating our
+	// store of trusted commits
+	StoreCommit(fc FullCommit) error
```

Comment:
StoreCommit is the only way we can have a trusted store, as the comments mention.

After I download a new validator set and approve it, I store it in the providers, so that I can use it as a reference for the next header I get.  If the code/comments wern't clear and this comment helped explain it, please update them so they make sense for someone besides me.